stable/database/files/nuosm

```diff
@@ -33,312 +39,499 @@ if [ $crashcount -ge 1 ]; then
   fi
 fi
 
-export NUOCMD DB_NAME NUODB_STORAGE_PASSWORDS_DIR
+export DB_NAME NUODB_STORAGE_PASSWORDS_DIR
 
 #=======================================
 # function - report an error and exit
 #
 function die() {
-  # cleanup
-  $NUOCMD set value --key $first_req --value '' --expected-value $HOSTNAME
-
   retval=$1
   shift
-  echo "$@"
+  error="$@"
+  [ -n "$wotIdid" ] && error="Error while ${wotIdid}: ${error}"
+  log "$error"
+
+  # cleanup
+  nuocmd set value --key $first_req --value '' --expected-value $HOSTNAME
+  nuocmd set value --key $startup_key/$DB_NAME --value '' --expected-value $HOSTNAME
```

Comment:
`first_req` and `startup_key` are defined later, which is weird but still legal in Bash (what isn't?). We should define those eagerly along with all the other variables above.

Reply:
I moved all KV store request keys at the beginning of the script.